controllers/release/release_adapter.go

```diff
@@ -113,10 +113,19 @@ func (a *Adapter) EnsureReleasePipelineRunExists() (results.OperationResult, err
 		return results.RequeueWithError(err)
 	}
 
-	var releaseStrategy *v1alpha1.ReleaseStrategy
+	var (
+		releasePlanAdmission *v1alpha1.ReleasePlanAdmission
+		releaseStrategy      *v1alpha1.ReleaseStrategy
+	)
 
 	if pipelineRun == nil {
-		releaseStrategy, err = a.getReleaseStrategyFromRelease()
+		releasePlanAdmission, err = a.getActiveReleasePlanAdmission()
+		if err != nil {
+			patch := client.MergeFrom(a.release.DeepCopy())
+			a.release.MarkInvalid(v1alpha1.ReleaseReasonTargetDisabledError, err.Error())
```

Comment:
Didn't we discuss that we would add a new operation to rule out this case before even trying to create the Pipeline? It also required changes in the `getActiveReleasePlanAdmission` function. With this code we are falling in the same thing. We are treating two errors (missing plan or disabled release) as the same one.

Reply:
No we are not because the error message is different. And we don't try to create the release pipeline until later on in this function so if the dev and managed workspace aren't properly linked, we never try to create a release pipeline

Reply:
One of the errors is about the ReleasePlanAdmission not being there (ReleaseReasonValidationError) while the other error is about the ReleasePlanAdmission containing a label which disables releases (ReleaseReasonTargetDisabledError).

Reply:
You want to make the release as invalid if one character is off in the application name of the releaseplan? I thought the entire point of this was all the releaseplan related errors should be a new type and not be marking the release as invalid, not just the auto-release label is disabled

Reply:
From the ticket
```
Acceptance Criteria
Releases marked with some new error types:
The managed ReleaseLink's target is NOT the dev ReleaseLink's namespace
The managed ReleaseLink and the dev ReleaseLink do not have the same application
The managed ReleaseLink is set to "auto-release": "false"
```

Reply:
Yeah, you wrote this particular ticket. In any case, you have a comment from me there mentioning that this should be a separated operation as we discussed during our talk discussing all the changes needed. Happy to go through that again if needed, but the summary is that a missing plan is a validation error while a plan disabling release is not an error.

Reply:
I created the ticket, then we groomed it and Scott added the acceptance criteria. So now your comment takes precedence over the team-agreed-upon acceptance criteria?